### PR TITLE
WebSocket unit test

### PR DESCRIPTION
**Pull request by neterror**
Unit test for upgrading to WebSocket and communication through it.

### CODE REVIEW

**Comment by neterror** [comment]
can you guide me how can I run these tests locally? I can't get why the test passes, but the pipeline in github hangs

**Comment by neterror** [comment]
I think it is OK now. The hanging problem was caused by improper streams cleanup.

**Review by j4qfrost — APPROVED** [review]
LGTM

File: packages/conduit/test/http/websocket_test.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        var hash = rx.toString();
+        if (hash == stopHash) {
+          break;
+        }
+
+        expect(messages[i++].hashCode.toString(),
+            rx.toString()); //check confirmation of each message
+      }
+    });
+  });
```
[the remark this answers is not in the document]
Comment (neterror) [reply packages/conduit/test/http/websocket_test.dart:124]: in which test did you add await socket.sink.close() ? I thought that all went smoothly when I added: await socket.sink.done. Confirmation, that the server has closed the socket. It didn't work only in the case, when single subscription channel was done after await (the second test). And in that case await socket.sink.done blocks - no incoming data can be received.

[the remark this answers is not in the document]
Comment (j4qfrost) [reply packages/conduit/test/http/websocket_test.dart:124]: The test succeeds in the regular unit tests, but fails when conduit generates a copy of itself to send into an isolate. The way you run those tests is detailed in the GitHub actions under conduit_tool. I don't know the specifics for why adding socket.sink.close() worked, but I'm guessing that deallocating socket for the isolate doesn't close before the test finishes. Adding the explicit call to close the socket before the isolate gets cleaned up makes it, so the the socket closes before the tests close.

[the remark this answers is not in the document]
Comment (neterror) [reply packages/conduit/test/http/websocket_test.dart:124]: I made a lot of changes to this test, trying to figure out what was the problem. Did you see the last version? I'm sorry that I didn't squash all my attempts into 1 commit. I can't see which line you added. Should I fix something now, or the master branch is OK?

[the remark this answers is not in the document]
Comment (j4qfrost) [reply packages/conduit/test/http/websocket_test.dart:124]: As long as the CI passes it should be fine.

**Pull request merged**